src/imginputfileconn.h

```diff
@@ -490,11 +490,44 @@ namespace dd
     {
       return _test_images.size();
     }
+
+    void get_data(const APIData &ad)
+    {
+      // check for raw cv::Mat
+      if (ad.has("data_raw_img"))
+	{
+	  if (ad.has("ids"))
+	    _ids = ad.get("ids").get<std::vector<std::string>>();
+	  if (ad.has("meta_uris"))
+	    _meta_uris = ad.get("meta_uris").get<std::vector<std::string>>();
+	  if (ad.has("index_uris"))
+	    _index_uris = ad.get("index_uris").get<std::vector<std::string>>();
+	  
```

Comment:
why not replace these 6 lines by InputConnectorStrategy::get_data(ad); ? 
changing the structure a bit would make things a bit more clear :
```
void get_data(ad)
{
   InputConnectorStrategy::get_data(ad);
   if (ad.has("data_raw_img"))
   {
      // all raw img stuff
   }
}
```

Reply:
Because if "data" is empty, since data is in "data_raw_img", the `get_data` call fails. 
I agree it's hacky though and that it'd be good to refactor the calls in transform so that they can be reused here.

Reply:
ok :)